### PR TITLE
Call finalize in Series.__array_ufunc__

### PR DESCRIPTION
**Pull request by TomAugspurger**
xref #28283

### CODE REVIEW

**Comment by jbrockmendel** [comment]
makes sense, looks mypy complaints

File: pandas/core/series.py [hunk 1]
```diff
@@ -739,7 +739,11 @@ def construct_return(result):
                     # GH#27198
                     raise NotImplementedError
                 return result
-            return self._constructor(result, index=index, name=name, copy=False)
+            # TODO: When we support multiple values in __finalize__, this should
```
Comment (jreback) [inline pandas/core/series.py:742]: I think i understand what you mean, but can you expand on this for readers

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
hmm can you merge master

**Comment by TomAugspurger** [comment]
This is being subsumed by https://github.com/pandas-dev/pandas/pull/36955

**Pull request closed without merging**